packages/deepbook-wrapper/sources/order.move

```
@@ -800,154 +920,146 @@ public(package) fun get_deep_plan(
 /// * `sui_in_wallet` - Amount of SUI available in user's wallet
 /// * `balance_manager_sui` - Amount of SUI available in user's balance manager
 ///
 /// # Returns
-/// * `FeePlan` - Struct containing:
+/// * `CoverageFeePlan` - Struct containing:
 ///   - Coverage fee amounts from wallet and balance manager
-///   - Protocol fee amounts from wallet and balance manager
 ///   - Whether user has sufficient funds to cover fees
 ///
 /// # Flow
 /// 1. Returns zero fee plan if pool is whitelisted or not using wrapper DEEP
-/// 2. Calculates total fee, coverage fee, and protocol fee
-/// 3. Returns zero fee plan if total fee is zero
+/// 2. Calculates coverage fee
+/// 3. Returns zero fee plan if coverage fee is zero
 /// 4. Returns insufficient fee plan if user lacks total funds
 /// 5. Plans coverage fee collection from available sources
-/// 6. Plans protocol fee collection from remaining funds
-public(package) fun get_fee_plan(
+public(package) fun get_coverage_fee_plan(
     use_wrapper_deep_reserves: bool,
     deep_from_reserves: u64,
     is_pool_whitelisted: bool,
-    protocol_fee_rate: u64,
     sui_per_deep: u64,
     sui_in_wallet: u64,
     balance_manager_sui: u64,
-): FeePlan {
+): CoverageFeePlan {
     // No fee for whitelisted pools or when not using wrapper DEEP
     if (is_pool_whitelisted || !use_wrapper_deep_reserves) {
-        return zero_fee_plan()
+        return zero_coverage_fee_plan()
     };
 
-    // Calculate fee based on order amount, including both protocol fee and deep reserves coverage fee
-    let (total_fee, coverage_fee, protocol_fee) = calculate_full_order_fee(
-        protocol_fee_rate,
+    // Calculate coverage fee based on order amount
+    let coverage_fee = calculate_deep_reserves_coverage_order_fee(
         sui_per_deep,
         deep_from_reserves,
     );
 
     // If no fee, return early
-    if (total_fee == 0) {
-        return zero_fee_plan()
+    if (coverage_fee == 0) {
```

One of your comments give me an idea about whether we need to silently return zero coverage fee plan here. Code execution here means at least one of the following:
1. SUI per DEEP is zero - something really flawed in DEEP/SUI price retrieving process.
2. DEEP from reserves are zero, but the pool is not whitelisted and wrapper DEEP reserves shouldn't be used (due to `if` above) - something flawed in DEEP planning process.

So should we maybe abort here? TBD